File: Selenium/Project/Activity9.java
package groupProject;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;
import org.testng.Assert;

public class Activity9 {
    public static void main(String[] args) {
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        // Open the page
        driver.get("https://alchemy.hguy.co/jobs/wp-admin/");
        WebElement userName= driver.findElement(By.id("user_login"));
        userName.sendKeys("root");
        WebElement password = driver.findElement(By.id("user_pass"));
        password.sendKeys("pa$$w0rd");
        WebElement loginButton= driver.findElement(By.id("wp-submit"));
        loginButton.click();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(5));
        WebElement jobLink = driver.findElement(By.xpath("//div[@class='wp-menu-name']/ancestor::div/ul/li[7]"));
        jobLink.click();
        WebElement addNew=driver.findElement(By.xpath("//*[text()='Add New']/ancestor::div/ul/li[7]/ul/li[3]"));
        addNew.click();
        WebDriverWait wait1 = new WebDriverWait(driver, Duration.ofSeconds(5));
        WebElement position=driver.findElement(By.xpath("//textarea[@id='post-title-0']"));
        position.sendKeys("TestLead");
        WebDriverWait wait2 = new WebDriverWait(driver, Duration.ofSeconds(5));
        WebElement email = driver.findElement(By.name("_application"));
        email.sendKeys("abhiram@cklabs.com");
        WebElement companyName = driver.findElement(By.name("_company_name"));
        companyName.sendKeys("IBM");
        WebElement companyWebsite = driver.findElement(By.id("_company_website"));
        companyWebsite.sendKeys("https://www.ibm.com");
        WebElement companyTagLine = driver.findElement(By.id("_company_tagline"));
        companyTagLine.sendKeys("IT Company");
        WebElement companyTwitter = driver.findElement(By.id("_company_twitter"));
        companyTwitter.sendKeys("@IBM");
        WebElement companyVideo = driver.findElement(By.id("_company_video"));
        companyVideo.sendKeys("https://www.ibm/video");
        WebElement jobExpiresDate = driver.findElement(By.name("_job_expires-datepicker"));
        jobExpiresDate.click();
        WebElement jobExpiryDate = driver.findElement(By.xpath("//a[text()='15']/ancestor::tr/td[5]"));
        jobExpiryDate.click();
        WebElement publishButton = driver.findElement(By.xpath("//button[contains(@class,'publish-button__button is-primary')]"));
        publishButton.click();
        WebElement publishButtonFinal = driver.findElement(By.xpath("//button[text()='Publish']"));
        publishButtonFinal.click();
        String pageTitle = driver.getTitle();
        Assert.assertEquals(pageTitle, "Add Job ‹ Alchemy Jobs — WordPress");
        
        driver.quit();
    }
}

File: Selenium/Project/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions
from pandas._testing.asserters import assert_equal

job_created = "Testing job"
# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Initialize the wait object
    wait = WebDriverWait(driver, 10)  
    # Navigate to the URL
    driver.get("https://alchemy.hguy.co/jobs/wp-admin")
    # Print the title of the page
    print("Page title is: ", driver.title)
    
    wait.until(expected_conditions.visibility_of_element_located((By.ID, "user_login")))
    
    driver.find_element(By.ID, "user_login").send_keys("root")
    driver.find_element(By.ID,"user_pass").send_keys("pa$$w0rd")
    
    driver.find_element(By.ID, "wp-submit").click()
    
    wait.until(expected_conditions.visibility_of_element_located((By.XPATH,"//*[@id='menu-posts-job_listing']/a/div[3]")))
    
    driver.find_element(By.XPATH, "//*[@id='menu-posts-job_listing']/a/div[3]").click()
    
    wait.until(expected_conditions.visibility_of_element_located((By.CLASS_NAME,"wp-heading-inline")))
    jobslist_page = driver.find_element(By.CLASS_NAME, "wp-heading-inline").text
    
    assert( jobslist_page == "Jobs")
    
    driver.find_element(By.CLASS_NAME,"page-title-action").click()
    
    
    wait.until(expected_conditions.visibility_of_element_located((By.XPATH, "//textarea[@id='post-title-0']")))
    driver.find_element(By.XPATH,"//textarea[@id='post-title-0']").send_keys(job_created)
    
    driver.find_element(By.ID,"_job_location").send_keys("Mysore")
    
    driver.find_element(By.ID,"_company_name").send_keys("IBM")
    
    driver.find_element(By.ID, "_company_tagline").send_keys("Indian Company")
    driver.implicitly_wait(10)
    
    driver.find_element(By.XPATH,"//button[contains(@class,'publish-button__button is-primary')]").click()
    
    wait.until(expected_conditions.visibility_of_element_located((By.XPATH,"//button[text()='Publish']")))
    driver.find_element(By.XPATH,"//button[text()='Publish']").click()
    
    wait.until(expected_conditions.visibility_of_element_located((By.LINK_TEXT, "Testing job")))
    driver.find_element(By.LINK_TEXT, "Testing job").click();
   # wait.until(expected_conditions.visibility_of_element_located((By.XPATH,"//*[@id='editor']/div/div/div[1]/div/div[2]/div[3]/div/div/div/div[2]/div/div[2]/div[2]/a")))
    #driver.find_element(By.XPATH,"//*[@id='editor']/div/div/div[1]/div/div[2]/div[3]/div/div/div/div[2]/div/div[2]/div[2]/a").click()
    
    wait.until(expected_conditions.visibility_of_element_located((By.CLASS_NAME, "entry-title")))
    
    jobcreated_title = driver.find_element(By.CLASS_NAME, "entry-title").text
    
    assert(jobcreated_title == job_created)
    
    
    
    driver.quit()
    
   
         
               